probe-rs-tools/src/bin/probe-rs/cmd/dap_server/server/debugger.rs

```diff
@@ -723,12 +745,31 @@ impl Debugger {
 
         download_options.progress = flash_progress;
 
-        let loader = build_loader(
+        let loader = match build_loader(
             &mut session_data.session,
             path_to_elf,
             self.config.flashing_config.format_options.clone(),
             None,
-        )?;
+        ) {
+            Ok(loader) => loader,
+            Err(error) => {
+                drop(download_options);
+                debug_adapter = match Rc::try_unwrap(rc_debug_adapter) {
+                    Ok(debug_adapter) => debug_adapter.into_inner(),
+                    Err(too_many_strong_references) => {
+                        let reference_error = DebuggerError::Other(anyhow!("Unexpected error while dereferencing the `debug_adapter` (It has {} strong references). Please report this as a bug.", Rc::strong_count(&too_many_strong_references)));
```

Comment:
I feel like this whole thing is safe enough for this to be an unwrap()

Reply:
If you are reffering to the `try_unwrap` ... that code was copied from code further down. I am not 100% on why this was originally added, but think it has something to do with protecting if something goes wrong in the flashing progress reporting.  I'm reluctant to remove it and risk the debug adapter panicking.